internal/app/progress.go

```diff
@@ -422,3 +425,69 @@ func generateHTML(progress runnerLocal.RunnerSyncProgress, errorLog *logger.Reve
 
 	return htmlOutput
 }
+
+// Convert the iface.Namespace key to a string ("db.col") to support JSON marshal
+func convertNamespaceMapToStringKeys(originalMap map[iface.Namespace]*iface.NamespaceStatus) map[string]*iface.NamespaceStatus {
+    newMap := make(map[string]*iface.NamespaceStatus)
+
+    for key, value := range originalMap {
+        newKey := fmt.Sprintf("%s.%s", key.Db, key.Col)
+        newMap[newKey] = value
+    }
+    return newMap
+}
+
+// Push updates using SSE server
+func progressUpdatesHandler(ctx context.Context, runner *runnerLocal.RunnerLocal, errorLog *logger.ReverseBuffer, w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "text/event-stream")
+	w.Header().Set("Cache-Control", "no-cache")
+	w.Header().Set("Connection", "keep-alive")
+	defer func() {
+        fmt.Fprintf(w, "data: complete\n\n")
```

Comment:
where do we parse this in the front end side. i noticed that usually we put json in here but this is just a string. i guess it is to signify the end of the stream, but hmm do we actually parse this or is this part of the sse spec?